datastore/prometheus.go

```diff
@@ -146,17 +160,96 @@ func (p *PrometheusExporter) updateMetrics(toPod PodEvent, req Request) {
 	}).Inc()
 }
 
+func (p *PrometheusExporter) handlePackets() {
+	for {
+		select {
+		case <-p.ctx.Done():
+			return
+		case pkt := <-p.pktChanBuffer:
```

Comment:
nit: You can check if the receive was due to channel close and bail out early
```suggestion
		case pkt, ok := <-p.pktChanBuffer:
		    if !ok {
		        return
		    }
```